tools/slang-test/main.cpp

```diff
@@ -74,9 +198,284 @@ struct Options
     Dictionary<TestCategory*, TestCategory*> excludeCategories;
 
     // By default we can test against all apis
-    int enabledApis = int(RenderApiFlag::AllOf);           
+    int enabledApis = int(RenderApiFlag::AllOf);
 };
-Options options;
+
+// Globals
+
+Options g_options;
+Dictionary<String, TestCategory*> g_testCategories;
+TestCategory* g_defaultTestCategory;
+
+// pre declare
+
+TestCategory* findTestCategory(String const& name);
+
+void append(const char* format, va_list args, StringBuilder& buf);
```

Comment:
Should these hypothetically move into `StringUtils` or `StringBuilder`? I guess doing it here avoids adding another dependency on the C stdlib in `core`, which would make it easier to make it independent of the C stdlib at a later date.